### PR TITLE
[wasm] Do not use link flags for bitcode compilation

### PR DESCRIPTION
**Pull request by radekdoulik**
This avoids errors like:

    /root/helix/work/correlation/build/wasm-shared/WasmApp.Common.targets(832,5): error : emcc: warning: linker setting ignored during compilation: 'EXPORT_ES6' [-Wunused-command-line-argument] [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]
    /root/helix/work/correlation/build/wasm-shared/WasmApp.Common.targets(832,5): error : emcc: warning: linker setting ignored during compilation: 'EXPORT_EXCEPTION_HANDLING_HELPERS' [-Wunused-command-line-argument] [/root/helix/work/workitem/e/publish/ProxyProjectForAOTOnHelix.proj]

### CODE REVIEW

File: src/mono/wasm/build/WasmApp.Common.targets [hunk 1]
```diff
@@ -804,7 +804,7 @@
     Outputs: file @(_WasmCompileBitcodeRsp
   -->
   <Target Name="_WasmWriteRspForCompilingBitcode">
-    <WriteLinesToFile Lines="@(_BitcodeLDFlags)" File="$(_WasmCompileBitcodeRsp)" Overwrite="true" WriteOnlyWhenDifferent="true" />
+    <WriteLinesToFile Lines="@(_EmccCommonFlags)" File="$(_WasmCompileBitcodeRsp)" Overwrite="true" WriteOnlyWhenDifferent="true" />
```
Comment (akoeplinger) [inline src/mono/wasm/build/WasmApp.Common.targets:807]: the comment above still references `_BitcodeLDFlags`.

I'm also lacking enough context to check why `_EmccCommonFlags` is enough 😄

Comment (radekdoulik) [reply src/mono/wasm/build/WasmApp.Common.targets]: updated the comment. also thinking more about it, I replaced the original _BitcodeLDFlags with _BitcodeCompileFlags. that way it is more clear and we still provide way to inject extra flags for BC compilation

Comment (radekdoulik) [reply src/mono/wasm/build/WasmApp.Common.targets]: the previously used `_EmccLDFlags` are based on `_EmccCommonFlags` plus linking specific flags. similar for the `_WasiClangLDFlags`

**Review by akoeplinger — APPROVED** [review]

**Comment by kg** [comment]
This seems like it may have caused an increase in file size for AOT and a resulting regression in AOT startup time. The regression is bigger for cold start than warm start, but both of them got worse around the time this landed

**Pull request merged**